cou/steps/plan.py

```diff
@@ -491,48 +524,65 @@ def _get_archive_data_steps(analysis_result: Analysis, args: CLIargs) -> list[Pr
     return []
 
 
-def _get_post_upgrade_steps(analysis_result: Analysis, args: CLIargs) -> list[PostUpgradeStep]:
-    """Get the post-upgrade steps.
+def _get_set_noout_steps(analysis_result: Analysis, args: CLIargs) -> list[PreUpgradeStep]:
+    """Get set noout steps.
 
     :param analysis_result: Analysis result
     :type analysis_result: Analysis
     :param args: CLI arguments
     :type args: CLIargs
-    :return: List of post-upgrade steps.
+    :return: List of pre-upgrade steps.
     :rtype: list[PreUpgradeStep]
     """
-    steps = []
-    if args.upgrade_group in {DATA_PLANE, None}:
-        steps.extend(_get_ceph_mon_post_upgrade_steps(analysis_result.apps_control_plane))
-
-    return steps
+    if args.set_noout and args.upgrade_group in {DATA_PLANE, None}:
+        return [
+            PreUpgradeStep(
+                description="Set ceph cluster 'noout' flag before data plane upgrade",
+                coro=ceph.osd_noout(analysis_result.model, enable=True),
```

Comment:
@chanchiwai-ray thanks for iterating on the function naming and signatures!  They're a lot easier to read now. :)